scripts/quests/sandoria/A_Timely_Visit.lua

```diff
@@ -0,0 +1,202 @@
+                end,
+                [21] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 9)
+                end,
+                [38] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 6)
+                end,
+                [78] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 5)
+                end,
+                [80] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 2)
+                end,
+                [87] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 4)
+                end,
+            },
+        },
+
+        [xi.zone.CHATEAU_DORAGUILLE] =
+        {
+            ['Halver'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 2 then
+                        return quest:progressEvent(112)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [112] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 3)
+                end,
+            },
+        },
+
+        [xi.zone.LA_THEINE_PLATEAU] =
+        {
+            ['Narvecaint'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 0 then
+                        return quest:progressEvent(0)
+                    elseif quest:getVar(player, 'Prog') == 1 then
+                        return quest:progressEvent(6) -- Additional Dialogue
+                    elseif quest:getVar(player, 'Prog') == 9 then
+                        return quest:progressEvent(1)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [0] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 1)
+                end,
+                [1] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 10)
+                end,
+            },
+        },
+
+        [xi.zone.JUGNER_FOREST] =
+        {
+            ['qm1'] =
+            {
+                onTrigger = function(player, npc)
+                    local mobID = GetMobByID(17203666)
+                    local time = false
+                    local check = false
+
+                    -- Check time
+                    if VanadielHour() <= 6 or VanadielHour() >= 18 then
+                        time = true
+                    end
+
+                    -- Check if mobs alive
+                    for i = 0, 1 do
+                        if mobID:isAlive() then
+                            check = false
+                        end
+                    end
+
+                    if quest:getVar(player, 'Prog') == 6 then
+                        if check and time then
```

Comment:
remove the mob alive check and use npcUtil
```lua
if time and npcUtil.popFromQM({mobs}) then
    return quest:message()
else
    return quest:message()
```

need those returns

Reply:
```suggestion
                        if
                            (VanadielHour() <= 6 or VanadielHour() >= 18) and
                            npcUtil.popFromQM(player, npc, ID.mob.TIMELY_VISIT, {claim = true, hide = 0})
                        then
                            return quest:messageText(ID.text.FOREBODING)
                        else
                            return quest:messageText(ID.text.PRETERNATURAL_FORCES)
                        end
```

Does this look better?

Reply:
Yes that would work out as well